library/Zend/Form/Element.php

```diff
@@ -83,6 +86,20 @@ public function init()
     }
 
     /**
+     * Remove the password before rendering if the form fails in order to avoid any security issue
+     *
+     * @param  FormInterface $form
+     * @return mixed
+     * @see    \Zend\Form\Element\Password::prepareElement
+     */
+    public function prepareElement(FormInterface $form)
+    {
+        if ('password' == $this->getAttribute('type')) {
+            $this->setValue('');
```

Comment:
This seems wrong to me: you don't know when `prepareElement` is going to be called. Instead, it would be better to filter out the value before rendering